.buildkite/premerge.steps.yaml

```diff
@@ -0,0 +1,59 @@
+common: &common
+  timeout_in_minutes: 60
+  retry:
+    automatic:
+        # These are designed to trap and retry failures because agent lost connection.
+        # Agent exits with -1 in this case.
+      - exit_status: -1
+        limit: 3
+      - exit_status: 255
+        limit: 3
+      - exit_status: 128
+        limit: 3
+
+linux_agent: &linux_agent
+  agents:
+    - "agent_count=1"
+    - "capable_of_building=worker-sdk"
+    - "environment=production"
+    - "permission_set=builder"
+    - "platform=linux"
+    - "scaler_version=2"
+    - "machine_type=half"
+    - "queue=${CI_LINUX_BUILDER_QUEUE:-v4-20-03-23-110204-bk9814-9f960b76}"
+  <<: *common
+
+macos_agent: &macos_agent
+  agents:
+    - "queue=${CI_MACOS_BUILDER_QUEUE:-macos_worker-sdk_17b57b101ca8ff76}"
+    - "environment=production"
+  <<: *common
+
+windows_agent: &windows_agent
+  agents:
+    - "agent_count=1"
+    - "capable_of_building=worker-sdk"
+    - "environment=production"
+    - "machine_type=half"
+    - "permission_set=builder"
+    - "platform=windows"
+    - "scaler_version=2"
+    - "queue=${CI_WINDOWS_BUILDER_QUEUE:-v4-20-03-25-183029-bk9912-014014bb}"
+  <<: *common
+
+steps:
+  - label: "build-linux"
+    command: "ci/build.sh"
+    <<: *linux_agent
+  - label: "build-macos"
+    command: "ci/build.sh"
+    <<: *macos_agent
+  - label: "build-windows"
+    command: "bash -c ci/build.sh"
+    <<: *windows_agent
+
+  - wait:
+
+  - label: build-steps-passed
+    command: echo Build steps passed!
+    <<: *linux_agent
```

Comment:
You don't need this fan-in / sink step.

Reply:
So I this is done to keep consistent with everything. I assume we have a fan-in / sink step there because we want to be able to change the steps without having to modify the repocop config. @RNabel probably knows more about why.

Reply:
In that case I'd suggest to simply block on the overall pipeline status check which should act as the overall `sink` step as it will only go green iff all steps have passed.

This is a little bit more subtle if you start to add manually unblocked wait steps (in which case if all steps are green until the `wait` block the overall pipeline is green) but seems to not be something that is likely to ever get in here.

Reply:
I didn't realize you can block on `all steps in green_state`, that would probably be the easier mechanism here. I'm guessing if we strove for super-maximal consistency the additional step would be better because it handles blocking steps, but I agree, for here the wait-on-all-steps mechanism is better (less complexity and gotchas) - unless we start using sentinel steps across the board, which seems unlikely.

Reply:
Sure, will change to block on the entire pipeline across the projects.